tools/integration-tests/tester/tests/consensus/consensus_test.go

```diff
@@ -0,0 +1,150 @@
+		peer2SeedBase58                      = "HUH4rmxUxMZBBtHJ4QM5Ts6s8DP3HnFpChejntnCxto2" // peerID iNvPFvkfSDp
+		peer1Pledged                         = 800000.0                                       // 40%
+		peer2Pledged                         = 400000.0                                       // 20%
+		actualGenesisTokenAmount      uint64 = 800000                                         // 40%
+		expectedCManaNode1AfterTxConf        = peer1Pledged + float64(actualGenesisTokenAmount)
+	)
+
+	var (
+		peer1IdentSeed = func() []byte {
+			seedBytes, err := base58.Decode(peer1SeedBase58)
+			require.NoError(t, err)
+			return seedBytes
+		}()
+
+		peer2IdentSeed = func() []byte {
+			seedBytes, err := base58.Decode(peer2SeedBase58)
+			require.NoError(t, err)
+			return seedBytes
+		}()
+	)
+
+	ctx, cancel := tests.Context(context.Background(), t)
+	defer cancel()
+	n, err := f.CreateNetwork(ctx, "test_simple_double_spend", 2, framework.CreateNetworkConfig{
+		StartSynced: true,
+		Faucet:      false,
+		Activity:    false,
+	}, func(peerIndex int, cfg config.GoShimmer) config.GoShimmer {
+		cfg.MessageLayer.Snapshot.File = "/assets/consensus_intgr_snapshot.bin"
+		cfg.UseNodeSeedAsWalletSeed = true
+		switch peerIndex {
+		case 0:
+			cfg.Autopeering.Seed = "base58:" + peer1SeedBase58
+			cfg.Seed = peer1IdentSeed
+		case 1:
+			cfg.Autopeering.Seed = "base58:" + peer2SeedBase58
+			cfg.Seed = peer2IdentSeed
+		}
+		return cfg
+	})
+	require.NoError(t, err)
+	defer tests.ShutdownNetwork(ctx, t, n)
+
+	var (
+		node1 = n.Peers()[0]
+		node2 = n.Peers()[1]
+
+		genesis1Wallet = createGenesisWallet(node1)
+		genesis2Wallet = createGenesisWallet(node2)
+
+		// 1G4a9TMsLE59BUbe76dacsFG63Ue2KN14orS5g4rPeNiK
+		node1TargetAddr = node1.Seed.Address(1)
+		// 1B64FizyBfUuycSKmAFADRiZo22SJj5pWrXoYvfY78q7L
+		node2TargetAddr = node2.Seed.Address(1)
+	)
+
+	// check consensus mana
+	require.EqualValues(t, peer1Pledged, tests.Mana(t, node1).Consensus)
+	require.EqualValues(t, peer2Pledged, tests.Mana(t, node2).Consensus)
+
+	// send conflicting txs on both
+	tx1 := sendConflictingTx(t, genesis1Wallet, node1TargetAddr, actualGenesisTokenAmount, node1, gof.Medium)
+	tx2 := sendConflictingTx(t, genesis2Wallet, node2TargetAddr, actualGenesisTokenAmount, node2, gof.Low)
+
+	// conflicting txs should have spawned branches
+	require.Eventually(t, func() bool {
+		res1, err := node1.GetTransactionMetadata(tx1.ID().Base58())
+		require.NoError(t, err)
+		res2, err := node2.GetTransactionMetadata(tx2.ID().Base58())
+		require.NoError(t, err)
+		return res1.BranchID != ledgerstate.MasterBranchID.String() &&
+			res2.BranchID != ledgerstate.MasterBranchID.String()
+	}, tests.Timeout, tests.Tick)
+
+	// we issue msgs on both nodes so the txs' GoF can change, given that they are dependent on their
+	// attachments' GoF. if msgs would only be issued on node 2 or 1, they weight would never surpass 50%.
```

Comment:
Is that true? If node 2 behaves honest it'd issue a message and if the branch is not liked it should set a like switch to the message of tx1 so that the branch would end up having 60% of weight, no? Either way issuing a bunch of messages on both nodes is also fine, just maybe update the comment.

Reply:
You are correct, that the **branch** of `tx1` would indeed gain 60% AW if only node 2 would issue messages and not both nodes. However, the GoF of a transaction is dependent on the attachment sites, therefore `tx1-msg` would not gain more AW since we only update message GoF based on markers. So in this case, future markers of `tx1-msg` would never have more than 20% AW, hence not allowing the AW of `tx1-msg` to gain over 40% and subsequent `tx1` to adjust its GoF.